### PR TITLE
fixtures: some type annotations, remove cyclic dependency

### PR DESCRIPTION
**Pull request by bluetech**
Please see the two commits.

### CODE REVIEW

File: src/_pytest/fixtures.py [hunk 1]
```diff
@@ -535,7 +546,7 @@ def _addfinalizer(self, finalizer: Callable[[], object], scope) -> None:
             finalizer=finalizer, colitem=colitem
         )
 
-    def applymarker(self, marker) -> None:
+    def applymarker(self, marker: Union[str, MarkDecorator]) -> None:
```
Comment (RonnyPfannschmidt) [inline src/_pytest/fixtures.py:549]: Should be plain marks as well

**Review by RonnyPfannschmidt — APPROVED** [review]
Looks lovely overall, thanks

File: src/_pytest/fixtures.py (hunk 1 above)
Comment (bluetech) [reply src/_pytest/fixtures.py:549]: Thanks for the review.

It seems that plain mark is not accepted here -- this function calls `self.node.add_marker(marker)` which starts with

```py
        from _pytest.mark import MARK_GEN

        if isinstance(marker, MarkDecorator):
            marker_ = marker
        elif isinstance(marker, str):
            marker_ = getattr(MARK_GEN, marker)
        else:
            raise ValueError("is not a string or pytest.mark.* Marker"
```

**Comment by aklajnert** [comment]
I always considered using dicts better than a chain of `elif`s. What's the rationale behind it?

**Comment by bluetech** [comment]
Good question, the reasons I changed it are:

- Due to the cyclic dependency it used to be initialized in a funky way in a hook. (There is still a cycle with `_pytest.python` so the second commit doesn't solve this.)

- With the if/elif chain the logic can be localized in one place which IMO is a bit easier to follow.

- I can use `assert_never` which means that if a new scope is ever added it will automatically trigger a mypy warning if a case for the new scope isn't added.

- Another reason I see now is that coverage doesn't work a "dict conditional" but works well with a chain of ifs.

**Comment by aklajnert** [comment]
I think that the `assert_never` can still be used with dicts.  
Anyway, all the reasons make sense, thanks for explaining.

**Review by nicoddemus — APPROVED** [review]

**Comment by nicoddemus** [comment]
Great work!

**Pull request merged**